src/gui/options_imp.cpp

```diff
@@ -583,6 +587,13 @@ int options_imp::getProxyType() const
     }
 }
 
+int options_imp::limitStep(int value)
+{
+    static const int pow10[] = {1, 1, 10, 100, 1000, 10000, 10000};
+    int log = static_cast<int>(log10(value));
```

Comment:
This value is already limited by `spinDownloadLimit->setMaximum(1000000)` in src/ui_options.h. I tried setting different potential problem values: -1, 0, 2000000, and the spinbox won't allow it.